src/core/utils/workspace.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { WORKSPACE_PATH_PREFIX } from './constants';
+import { IBasePath } from '../public';
+
+export const getWorkspaceIdFromUrl = (url: string, basePath?: string): string => {
+  const regexp = new RegExp(`^${basePath || ''}\/w\/([^\/]*)`);
+  const urlObject = new URL(url);
+  const matchedResult = urlObject.pathname.match(regexp);
+  if (matchedResult) {
+    return matchedResult[1];
+  }
+
+  return '';
+};
+
+export const cleanWorkspaceId = (path: string) => {
+  return path.replace(/^\/w\/([^\/]*)/, '');
+};
+
+export const formatUrlWithWorkspaceId = (url: string, workspaceId: string, basePath: IBasePath) => {
```

Comment:
Do we handle the edge cases here? like as `workspaceId` is an empty string or `undefined` when `basePath` is not provided or is empty string?

Reply:
I think that's fine because we should handle the edge cases as what function declares and the basePath has to be present. And the test cases for formatUrlWithWorkspaceId has covered the case when basePath is present but basePath.serverBasePath is empty.